### PR TITLE
[bridgeworld-stats] attempt to fix pilgrimage stat counts

### PR DESCRIPTION
**Pull request by alecananian**
- Updates packages
- Attempts to fix an issue seen in the Pilgrimage stats with some count clean-up

@wyze any other reason you can see for this mismatch?

```
{
  "data": {
    "pilgrimageStat": {
      "pilgrimagesStarted": 4359,
      "pilgrimagesFinished": 4397,
      "activeAddressesCount": 19,
      "allAddressesCount": 1680
    }
  }
}
```

Finished count is somehow more than started count in the synced production subgraph 🤔

### CODE REVIEW

File: subgraphs/bridgeworld-stats/src/mappings/pilgrimage.ts [hunk 1]
```diff
@@ -13,11 +13,10 @@ import {
 
 export function handlePilgrimagesStarted(event: PilgrimagesStarted): void {
   const params = event.params;
-  const tokenIds = params._ids1155;
   const tokenAmounts = params._amounts1155;
 
   let pilgrimagesCount = 0;
-  for (let i = 0; i < tokenIds.length; i++) {
+  for (let i = 0; i < tokenAmounts.length; i++) {
```
Comment (wyze) [inline subgraphs/bridgeworld-stats/src/mappings/pilgrimage.ts:19]: This change doesn't really matter as the ids and amounts array will always be the same length.

File: subgraphs/bridgeworld-stats/src/mappings/pilgrimage.ts [hunk 2]
```diff
@@ -52,11 +51,12 @@ export function handlePilgrimagesStarted(event: PilgrimagesStarted): void {
 export function handlePilgrimagesFinished(event: PilgrimagesFinished): void {
   const params = event.params;
   const tokenIds = params._tokenIds;
+  const pilgrimagesCount = params._finishedPilgrimageIds.length;
 
   const stats = getTimeIntervalPilgrimageStats(event.block);
   for (let i = 0; i < stats.length; i++) {
     const stat = stats[i];
-    stat.pilgrimagesFinished += tokenIds.length;
+    stat.pilgrimagesFinished += pilgrimagesCount;
```
Comment (wyze) [inline subgraphs/bridgeworld-stats/src/mappings/pilgrimage.ts:59]: Again, the tokenIds and pilgrimageId lengths will always be the same.

**Comment by wyze** [comment]
I don't think you can go off length of pilgrimageIds. Here is the code:

```
        uint256[] memory _minted721s = new uint256[](_inProgressPilgrimages.length);
        uint256 _minted721sIndex = 0;

        uint256[] memory _finishedPilgrimages = new uint256[](_inProgressPilgrimages.length);
        uint256 _finishedPilgrimagesIndex = 0;

        for(uint256 i = 0; i < _inProgressPilgrimages.length; i++) {
            uint256 _pilgrimageId = _inProgressPilgrimages[i];

            uint256 _tokenId721 = _returnFromPilgrimage(_pilgrimageId);
            if(_tokenId721 != 0) {
                _minted721s[_minted721sIndex] = _tokenId721;
                _minted721sIndex++;
                _finishedPilgrimages[_finishedPilgrimagesIndex] = _pilgrimageId;
                _finishedPilgrimagesIndex++;
                userToPilgrimagesInProgress[msg.sender].remove(_pilgrimageId);
            }
        }
```

You can see the length is set up front, so it is fixed. But if the pilgrimage is not complete, then it wont add an entry to the array.

File: subgraphs/bridgeworld-stats/src/mappings/pilgrimage.ts (hunk 1 above)
Comment (alecananian) [reply subgraphs/bridgeworld-stats/src/mappings/pilgrimage.ts:19]: That's what I figured, but better to make it explicitly pulling from the `tokenAmounts` here

**Comment by alecananian** [comment]
Interesting find there with the finished pilgrimage IDs. I'm assuming they'd come in as zeroes... going to look for a transaction where this is happening so i can confirm.

**Comment by alecananian** [comment]
Added a filter for only including non-zero pilgrimage IDs in finished count

**Review by wyze — APPROVED** [review]

**Pull request merged**